src/gc/gc.d

```diff
@@ -53,12 +54,67 @@ private alias BlkInfo = core.memory.GC.BlkInfo;
 
 version (GNU) import gcc.builtins;
 
-debug (PRINTF_TO_FILE) import core.stdc.stdio : fprintf, fopen, fflush, FILE;
-else                   import core.stdc.stdio : printf; // needed to output profiling results
+debug (PRINTF_TO_FILE) import core.stdc.stdio : sprintf, fprintf, fopen, fflush, FILE;
+else                   import core.stdc.stdio : sprintf, printf; // needed to output profiling results
 
 import core.time;
 alias currTime = MonoTime.currTime;
 
+// high performance time measurements without additional translations
+alias MonoTimeNative = long;
+alias DurationNative = long;
+
+version(OSX) extern(C) ulong mach_absolute_time() nothrow @nogc;
+
+@property MonoTimeNative currTimeNative() nothrow
+{
+    version(Windows)
+    {
+        import core.sys.windows.windows;
+        long ticks;
+        if(QueryPerformanceCounter(&ticks) == 0)
+        {
+            // This probably cannot happen on Windows 95 or later
+            assert(0, "Call to QueryPerformanceCounter failed.");
+        }
+        return ticks;
+    }
+    else version(OSX)
+    {
+        return mach_absolute_time();
+    }
+    else version(Posix)
+    {
+        import core.sys.posix.time;
+        import core.sys.posix.sys.time;
+
+        timespec ts;
+        if(clock_gettime(CLOCK_MONOTONIC, &ts) != 0)
+            assert(0, "Call to clock_gettime failed.");
+
+        return ts.tv_sec * 1_000_000_000L + ts.tv_nsec;
+    }
+    else
+        static assert(false, "no native timer function");
+}
+
+MonoTime toMonoTime(MonoTimeNative tm)
+{
+    version(Windows)
+        return MonoTime(tm);
+    else version(OSX)
+        return MonoTime(tm);
+    else version(Posix)
+        return MonoTime(convClockFreq(tm, 1_000_000_000L, MonoTime.ticksPerSecond));
```

Comment:
I've replaced the code duplication with a runtime check for MonoTime._ticksPerSecond.